modules/core/src/main/scala/org/scalasteward/core/maven/MavenAlg.scala

```diff
@@ -0,0 +1,209 @@
+
+      def maybeIgnoreOptsFiles[A](dir: File)(fa: F[A]): F[A] =
+        if (config.ignoreOptsFiles) ignoreOptsFiles(dir)(fa) else fa
+
+      def ignoreOptsFiles[A](dir: File)(fa: F[A]): F[A] =
+        fileAlg.removeTemporarily(dir / ".jvmopts") {
+          fa
+        }
+
+//      override def getUpdatesForRepo(repo: Repo): F[List[Update.Single]] =
+//        for {
+//          repoDir <- workspaceAlg.repoDir(repo)
+//          minorUpdatesRawLines <- exec(mvnCmd(command.MvnMinorUpdates), repoDir)
+//          minorUpdates = parseUpdates(minorUpdatesRawLines)
+//          majorUpdatesRawLines <- exec(mvnCmd(command.MvnDepUpdates), repoDir)
+//          majorUpdates = parseUpdates(majorUpdatesRawLines)
+//          pluginUpdatesRawLines <- exec(mvnCmd(command.PluginUpdates), repoDir)
+//          pluginUpdates = parseUpdates(pluginUpdatesRawLines)
+//          updates = Update.groupByArtifactIdName(minorUpdates ++ majorUpdates)
+//          result = updates ++ pluginUpdates
+//        } yield result
+
+      override def runMigrations(repo: Repo, migrations: Nel[Migration]): F[Unit] =
+        F.unit //fixme:implement
+    }
+
+  private def mvnCmd(commands: String*): Nel[String] =
+    Nel.of("mvn", commands.flatMap(_.split(" ")): _*)
+
+  def removeNoise(s: String): String = s.replace("[INFO]", "").trim
+
+  //fixme: rewrite with atto
+  def parseDependencies(lines: List[String]): List[Dependency] = {
+    val pattern = """(.*):(.*):jar:(.*):compile""".r
+    lines
+      .map(removeNoise)
+      .map(_.trim)
+      .map { s =>
+        Try {
+          val pattern(groupId, artifactIdCross, currentVersion) = s
+          val artifactId = artifactIdCross.split("_") //fixme: check if it's scala binary
+          new Dependency(
+            GroupId(groupId),
+            ArtifactId(artifactId(0), artifactIdCross),
+            currentVersion
+          )
+        }.toOption // TODO: this doesn't catch exceptions thrown by Try, if any
+      // TODO: does regex throw exceptions if there are no matches?
+      // todo: add logger and log the exceptions
+      }
+      .collect { case Some(x) => x }
+  }
+
+//  def parseUpdates(lines: List[String]): List[Update.Single] =
+//    lines
+//      .map(removeNoise)
+//      .map(_.trim)
+//      .map { s =>
+//        MavenParser.parse(s).toOption
+//      }
+//      .collect { case Some(x) => x }
+}
+
+object MavenParser {
+
+  private val dot: Parser[Char] = char('.')
+  private val arrow = char('-') ~ char('>')
+
+  case class Version(major: Int, minor: Int, patch: Option[Int]) {
+    override def toString: String = {
+      val x = (major :: minor :: Nil) ++ patch.toList
+      x.map(_.toString).mkString(".")
+    }
+  }
+
+  private val version: Parser[Version] = {
+    val version3args: Parser[Version] =
+      for {
+        a <- int <~ dot
+        b <- int <~ dot
+        c <- int
+      } yield Version(a, b, Some(c))
+
+    val version2args: Parser[Version] =
```

Comment:
what is the difference with version3args ?

Reply:
@tyrcho Hi, this bit is not actually used and I just removed it, currently only `override def getDependencies(repo: Repo): F[List[Scope.Dependencies]]` needs to be implemented and `def parseDependencies(lines: List[String]): List[Dependency]` method rewritten with atto (and the output type changed to `List[Scope.Dependencies]`) -  will probably require an output of some other maven commands to additionally get `resolvers: List[Resolver]` 
I've been quite busy at work lately and kind of neglected the PR but I'll try to finally finish the work